allauth/socialaccount/providers/tiktok/views.py

```diff
@@ -0,0 +1,48 @@
+from allauth.socialaccount.adapter import get_adapter
+from allauth.socialaccount.providers.oauth2.views import (
+    OAuth2Adapter,
+    OAuth2CallbackView,
+    OAuth2LoginView,
+)
+from allauth.socialaccount.providers.tiktok.client import TikTokOAuth2Client
+from allauth.socialaccount.providers.tiktok.scope import TikTokScope
+
+
+class TikTokOAuth2Adapter(OAuth2Adapter):
+    provider_id = "tiktok"
+    access_token_url = "https://open.tiktokapis.com/v2/oauth/token/"
+    authorize_url = "https://www.tiktok.com/v2/auth/authorize/"
+    # https://developers.tiktok.com/doc/tiktok-api-v2-get-user-info/
+    profile_url = "https://open.tiktokapis.com/v2/user/info/"
+    client_class = TikTokOAuth2Client
+    scope_delimiter = ","
+
+    def get_query_fields(self):
+        fields = ""
+        if TikTokScope.user_info_basic in self.get_provider().get_scope():
+            fields = "open_id,display_name,avatar_url"
+        if TikTokScope.user_info_profile in self.get_provider().get_scope():
+            fields += ",username,profile_deep_link"
```

Comment:
If `user_info_basic` is not in scope, but `user_info_profile` is, then `fields` will start using a `","`.

Reply:
I don't think you would make a request without basic scope but fair point